tensorly/regression/cp_plsr.py

```diff
@@ -0,0 +1,230 @@
+from ..tenalg import khatri_rao, mode_dot, multi_mode_dot
+from ..cp_tensor import CPTensor, cp_to_tensor, cp_to_vec
+from ..decomposition import tucker
+from .. import backend as T
+from .. import unfold
+
+# Author: Cyrillus Tan, Jackson Chin, Aaron Meyer
+
+# License: BSD 3 clause
+
+
+class CP_PLSR:
+    """CP tensor regression
+
+        Learns a low rank CP tensor weight
+
+    Parameters
+    ----------
+    n_components : int
+        rank of the CP decomposition of the regression weights
+    tol : float
+        convergence value
+    n_iter_max : int, optional, default is 100
+        maximum number of iteration
+    random_state : None, int or RandomState, optional, default is None
+    verbose : bool, default is False
+        whether to be verbose during fitting
+    """
+
+    def __init__(
+        self, n_components, tol=1.0e-7, n_iter_max=100, random_state=None, verbose=False
+    ):
+        self.n_components = n_components
+        self.tol = tol
+        self.n_iter_max = n_iter_max
+        self.random_state = random_state
```

Comment:
It looks like there is no random number generation. If this is the case then you can remove random_state.